src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -63,27 +81,24 @@ public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidCon
                                        ItemComponent itemComponent) {
         if (fluidContainer.fluidType == null || fluidContainer.volume < fluidContainer.maxVolume) {
             Vector3f location = event.getInstigatorLocation();
-            Vector3f direction = new Vector3f(event.getDirection());
-            direction.normalize();
-            for (int i = 0; i < 3; i++) {
-                location.add(direction);
+            Vector3f direction = new Vector3f(event.getDirection()).normalize();
 
-                Block block = worldProvider.getBlock(new Vector3i(location, 0.5f));
-                if (block.isLiquid() ) {
-                    EntityRef owner = item.getOwner();
-                    final EntityRef removedItem = inventoryManager.removeItem(owner, event.getInstigator(), item, false, 1);
-                    if (removedItem != null && block.isWater()) {
-                        // Set the contents of this fluid container and fill it up to max capacity.
-                        FluidUtils.setFluidForContainerItem(removedItem, "Fluid:Water",
-                                removedItem.getComponent(FluidContainerItemComponent.class).maxVolume);
+            Optional<Block> block = getLiquidInReach(location, direction, 3);
+            if (block.isPresent() && block.get().isLiquid()) {
```

Comment:
The `isLiquid()` property is guaranteed by `getLiquidInReach`, thus this check can be simplified.

```suggestion
            Optional<Block> block = getLiquidInReach(location, direction, 3);
            if (block.isPresent()) {
```

Reply:
Instead of using the pattern 

```java
Optional<Block> block = ....;
if (block.isPresent()) {
	... block.get()....
}
```

you could use [`Optional::ifPresent`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-) here:

```suggestion
            getLiquidInReach(location, direction, 3).ifPresent(block -> {
```